cmd/kubernetes/kubernetes.go

```diff
@@ -145,12 +146,24 @@ func runControllerManager(machineList []string, cl *client.Client, nodeMilliCPU,
 	controllerManager.Run(controller.DefaultSyncPeriod)
 }
 
+// runServiceAccountController starts controllers to manage service accounts and tokens
+func runServiceAccountController(cl *client.Client) {
+	// TODO: generate kubeconfig? pods might need ca bundle, hostname, etc
+	// TODO: generate signed token
+	tokenGenerator := serviceaccount.TokenGeneratorFunc(func(serviceAccount *api.ServiceAccount, secret *api.Secret) (string, error) {
+		return fmt.Sprintf("serviceaccount:%s:%s:%s:%s", serviceAccount.Namespace, serviceAccount.Name, serviceAccount.UID, secret.UID), nil
+	})
+	serviceaccount.NewTokensController(cl, serviceaccount.DefaultTokenControllerOptions(tokenGenerator)).Run(util.NeverStop)
+	serviceaccount.NewServiceAccountsController(cl, serviceaccount.DefaultServiceAccountControllerOptions()).Run(util.NeverStop)
+}
+
 func startComponents(etcdClient tools.EtcdClient, cl *client.Client, addr net.IP, port int) {
 	machineList := []string{"localhost"}
 
 	runApiServer(etcdClient, addr, port, *masterServiceNamespace)
 	runScheduler(cl)
 	runControllerManager(machineList, cl, *nodeMilliCPU, *nodeMemory)
+	runServiceAccountController(cl)
```

Comment:
I'm having a conversation with myself.  runControllerManager runs the same controllers as controllermanager but they are not the same thing.  

I suppose the question then is why a separate run method for your controller instead of rolling it into runControllerManager (which ideally wouldn't have duplication with controllermanager.go)?